Sources/KavaPlugin+AnalyticsPluginProtocol.swift

```diff
@@ -0,0 +1,218 @@
+                    strongSelf.sendAnalyticsEvent(event: KavaEventType.playRequest)
+                    
+                    if strongSelf.sentPlaybackPoints[KavaEventType.playReached100Percent] == true {
+                        strongSelf.sendAnalyticsEvent(event: KavaEventType.replay)
+                    }
+                })
+                
+            case let e where e.self == PlayerEvent.pause:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("pause event: \(event)")
+                    strongSelf.kavaData.isPaused = true
+
+                    strongSelf.sendAnalyticsEvent(event: KavaEventType.pause)
+                    strongSelf.reportView()
+                    strongSelf.stopViewTimer()
+                })
+                
+            case let e where e.self == PlayerEvent.playing:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("playing event: \(event)")
+                    if strongSelf.isFirstPlay {
+                        strongSelf.isFirstPlay = false
+                        strongSelf.sendAnalyticsEvent(event: KavaEventType.play)
+                    } else if strongSelf.kavaData.isPaused {
+                        strongSelf.sendAnalyticsEvent(event: KavaEventType.resume)
+                    }
+                    
+                    strongSelf.kavaData.isPaused = false
+                    strongSelf.setupViewTimer()
+                })
+                
+            case let e where e.self == PlayerEvent.seeking:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    PKLog.debug("seeking event: \(event)")
+                    
+                    guard let strongSelf = self else { return }
+                    
+                    if let seekPosition = event.targetSeekPosition {
+                        strongSelf.kavaData.targetSeekPosition = Double(truncating: seekPosition)
+                    }
+                    
+                    strongSelf.sendAnalyticsEvent(event: KavaEventType.seek)
+                })
+                
+            case let e where e.self == PlayerEvent.sourceSelected:
+                self.messageBus?.addObserver(self, events: [e.self], block: { [weak self] (event) in
+                    guard let strongSelf = self else { return }
+                    PKLog.debug("sourceSelected event: \(event)")
+                    strongSelf.kavaData.selectedSource = event.mediaSource
+                    strongSelf.updateDeliveryType(mediaFormat: (strongSelf.kavaData.selectedSource?.mediaFormat)!)
+                    // Reset flags when source was changed
+                    strongSelf.resetPlayerFlags()
```

Comment:
better reset state on life cycle events - we should set it on media update life cycle hook